src/Symfony/Component/Serializer/Normalizer/CheckCircularReferenceNormalizer.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Exception\CircularReferenceException;
+use Symfony\Component\Serializer\SerializerAwareInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Handle circular references.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+final class CheckCircularReferenceNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface, NormalizerAwareInterface, DenormalizerAwareInterface, SerializerAwareInterface
+{
+    public const CIRCULAR_REFERENCE_LIMIT = 'circular_reference_limit';
```

Comment:
I'm not sure to duplicate const.
I think, for the moment. you can referrer to original const as :
`public const CIRCULAR_REFERENCE_LIMIT = AbstractNormalizer::CIRCULAR_REFERENCE_LIMIT`
When the deprecated on AbstractNormalizer will be add, we will can revert the relation with 
`public const CIRCULAR_REFERENCE_LIMIT = CheckCircularReferenceNormalizer::CIRCULAR_REFERENCE_LIMIT` on `AbstractNormalizer`

Reply:
@jewome62 The const value can't be changed for BC, so it's alright.